### DISCUSSION TITLE
Selecting gpt 3.5 16k?

### DISCUSSION THREAD

**Discussion by w0010** [brianpetro/obsidian-smart-connections#243] [General]
Is it possible to select the new gpt-3.5-turbo-16k? The drop-down menu only has gpt-3.5-turbo and gpt-4.

**Comment by brianpetro** [comment]
@w0010 I just pushed an update containing the new model. 

Let me know how it works for you.

**Reply by brianpetro** [reply]
It requires changing the model in the settings.

**Reply by liuba1223** [reply]
I'm sorry to bother you. I updated the latest version of the plugin and selected the model option as 16k, but when processing my 3000-word note, it still reported an error: API Error. See console logs for details.

**Reply by w0010** [reply]
> @w0010 I just pushed an update containing the new model.
> 
> Let me know how it works for you.

Awesome! Got it changed in the model settings.

Well, it works sometimes. If I ask it for a summary on a specific note, it'll respond. But if I'm more vague and refer to an idea that spans all my notes, (Imagine the following `20` [rather than `1`] notes...) it comes back with: *API Error. See console logs for details.* I've turned on `log_render`, but I cannot find where the log lives to post it.

**Reply by brianpetro** [reply]
@w0010 

Sounds like I might've missed on some token calculation.

In the developer console you should be able to some logs that contain something like "max tokens" when you send a message.

Let me know if you see any patterns in those logs when you get the error.

Thanks for your help.
🌴

**Reply by w0010** [reply]
> @w0010
> 
> Sounds like I might've missed on some token calculation.
> 
> In the developer console you should be able to some logs that contain something like "max tokens" when you send a message.
> 
> Let me know if you see any patterns in those logs when you get the error.
> 
> Thanks for your help. 🌴

Perhaps this is what you are looking for:
```
{
  "error": {
    "message": "This model's maximum context length is 16385 tokens. However, you requested 18982 tokens (8998 in the messages, 9984 in the completion). Please reduce the length of the messages or completion.",
    "type": "invalid_request_error",
    "param": "messages",
    "code": "context_length_exceeded"
  }
}
```

**Reply by S-Wil** [reply]
I have the same issue - in fact no matter which model I select I get an error saying I have submitted more tokens than the model can handle:
"message": "This model's maximum context length is 16385 tokens. However, you requested 19897 tokens (9978 in the messages, 9919 in the completion). Please reduce the length of the messages or completion.",
"message": "This model's maximum context length is 4097 tokens. However, you requested 5137 tokens (2691 in the messages, 2446 in the completion). Please reduce the length of the messages or completion.",

**Reply by brianpetro** [reply]
@w0010 @S-Wil thanks for your help in solving this.

Are your notes in English? What about use of other "special characters" in your notes?

I'll also be reviewing the token estimate calculations today to see if I overlooked something.

I'll probably also add another setting that scales down the number of max tokens, which will be necessary in the case of non-english languages and special character usage.

Thanks again for your help
Brian 🌴

**Reply by S-Wil** [reply]
All English for me.  To provide some more context in case it's relevant: my notes are mostly all text.  Minimal formatting, minimal code blocks, etc.  About the only thing I can think of is that I do embed images (sometimes url links but mostly images locally stored in the vault).

**Reply by brianpetro** [reply]
@w0010 @S-Wil I just fixed a bug in the way the tokens are calculated. 

This should resolve the issue.

Please do let me know if it keeps happening.

Thanks for your help in solving this
🌴

**Comment by S-Wil** [comment]
hmm now I am getting a different error:
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/51426105/4225d3df-f8e7-497d-90a3-31b914f6fa45)

**Reply by brianpetro** [reply]
Does it keep happening?

I traced this, and it looks like a prior request, `get_context_hyde`, failed to produce the expected output.

Thanks for your help in figuring this out
Brian 🌴

**Comment by S-Wil** [comment]
I have uninstalled your plugin, restarted obsidian several times, replaced the API key, created new notes, etc. and yes it keeps happening.  I haven't rebooted yet but I'll get to that too.  This message pops up in the UI when I send:
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/51426105/572addb4-2c58-44af-a6c2-b6e1123cf908)

**Reply by brianpetro** [reply]
Can you give me an example message you are sending?

Have you tried changing the `Smart Chat Model` in the settings?

Thanks
🌴

**Comment by S-Wil** [comment]
Something probably unrelated but there's a bug in the text entry area where it shows this sometimes:
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/51426105/abb39280-712e-42c2-9802-5cbf1465d50c)
If I click on the tickbox it changes to the send button and then I can send it.

**Reply by brianpetro** [reply]
That's the "stop streaming response" button, similar to what's available in ChatGPT.

It might be getting stuck because of the errors, though.

**Comment by S-Wil** [comment]
The sample message is the one in the screenshot above.  Weird - I changed models and I received one response.  Then with the next prompt another error:
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/51426105/935ccc64-53fd-4716-86e2-81d752e03913)

**Reply by brianpetro** [reply]
@S-Wil are you updated to `1.6.28`?

I tried to address this issue in that update, however there could still be edge cases.

**Comment by S-Wil** [comment]
Doh - I updated about an hour ago but yes when I look at the version it was still on 1.6.27.  I have updated now and it seems to be working.  Sorry for the confusion.  Thanks for your help with this.  Great work with the plugin!

**Reply by brianpetro** [reply]
Thanks for following up!
🌴
